### ISSUE TITLE
Firefox web extension stop working after log in with device

### ISSUE THREAD

**Issue by IamBennJ** [bitwarden/clients#4878]
### Steps To Reproduce

1. Go to the firefox extension
2. Click on 'Login with device' ( not master password )
3. Close the extension
4. Open the extension again

### Expected Result

Normal extension's menu with the 5 icons at the bottom level ect..

### Actual Result

The extension's menu is dark, no buttons, nothing ( see screenshot )

### Screenshots or Videos

![Capture d’écran (1)](https://user-images.githubusercontent.com/82703909/221536945-f094dcd1-366a-4b95-b45f-0f0ead38ee56.png)

### Additional Context

Everything works normally when login with master password. When login with device, the extension works the first time ( immediately after login ) but stops working after closing the extension. When opening the extension again, ithere is no buttons nor icons, nothing.

### Operating System

Windows

### Operating System Version

10.0.22621 Build 22621 ( Latest update of Windows 11 on 27/02/2023 )

### Web Browser

Firefox

### Browser Version

110.0 (64 bits)

### Build Version

2023.2.1

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by Greenderella** [comment]
Hi there,

I attempted to reproduce your issue and was unable to do so. If you have performed a clean installation of the browser extension already by removing any leftover data (https://bitwarden.com/help/data-storage/#on-your-local-machine), I suggest you try to refresh your Firefox's profile following their instructions: https://support.mozilla.org/en-US/kb/refresh-firefox-reset-add-ons-and-settings

We use GitHub issues as a place to track bugs and other development related issues. If your issue persists, please write us back using our contact form, so we can continue troubleshooting: https://bitwarden.com/contact/ 

You can include a link to this issue in the message content.

Alternatively, you can also search for an answer in our help documentation (https://bitwarden.com/help/) or get help from other Bitwarden users on our community forums (https://community.bitwarden.com/c/support/). 

The issue here will be closed.

Thanks!

**Comment by katchy3132** [comment]
I am having the same issue. With password login, no issues but with login with device I get a result like above. I reinstalled the extension a few times and made sure there was no leftover data. I have not done a refresh since I do not want to go through the trouble of reinstalling all my extensions and customizations but if that's the fix I'll deal with it . 

When I reinstall the extension, I login on the sidebar and then get the following with the extension.  Restarting Firefox shows the regular dropdown. 
![image](https://user-images.githubusercontent.com/3707138/222391797-d16a5f65-dc38-4a14-acf5-777e13d59658.png)

**Comment by jovabel** [comment]
Same Issue over here. Refreshed firefox. Cleaned up left over data. Reinstalled the extension. Bot nothing helps.
When logging in with the master password, everything works.
When logging in with a device, I get the same grey box as is mentioned in the other posts.

Using Windows 10 Pro, 22H2, Build 19045.2604
Firefox 110.0.1 (64-bit)
Bitwarden Extension for Firefox 2023.2.1

**Comment by iCodeHead** [comment]
Same issue, after using login with device, clicking on the Firefox Bitwarden extension shows up blank.
I'm running Bitwarden 2023.2.1, Firefox version 110.0.1 (64-bit) with both Windows 8.1 and Windows 10.
This issue does not occur on Chrome browsers.
This is easy to reproduce using the debugger, if it helps the devs, I see several errors like the following. 
ERROR Error: Uncaught (in promise): TypeError: can't access dead object
    l background.js:1
ERROR TypeError: can't access dead object
    updateSession session-syncer.ts:106
Let me know if you need the full log.

**Comment by Galactic-Hero** [comment]
Same problem here. I was able to get the extension to work again by disabling it, turning off "allow device login" on my iphone app, then reenabling the extension.

**Comment by shanedk** [comment]
I have exactly the same issue. Disabling/reenabling allowed me to lock back in, but it became blank immediately after.

**Comment by DeveloperRob** [comment]
I am experiencing this issue myself, and I can replicate the issue in a clean install of Firefox in Windows Sandbox (so a clean, new install still causes the issue).

Comparing the `Multiprocess Browser Console` between logging in with device and logging in normally shows a number of errors related to a dead object. The UUID matches the extension UUID in `about:debugging#/runtime/this-firefox`. 

```
Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. [background.js:1](moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/background.js)
    sendMessage moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/background.js:1
ERROR Error: Uncaught (in promise): TypeError: can't access dead object
l@moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/background.js:1:419069
o@moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/background.js:1:418968
    P moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    P moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    C moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    invokeTask moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    onInvokeTask Angular
    invokeTask moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    runTask moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    b moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
[vendor-angular.js:2:318527](moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/vendor-angular.js)
ERROR TypeError: can't access dead object
    P moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    O moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    A moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    l moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/main.js:1
    r moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/main.js:1
    invoke moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    onInvoke Angular
    invoke moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    run moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    C moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    invokeTask moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    onInvokeTask Angular
    invokeTask moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    runTask moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    b moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
[vendor-angular.js:2:318527](moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/vendor-angular.js)
ERROR TypeError: can't access dead object
    updateSession moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/main.js:1
    IA moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/main.js:1
    t moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    IA moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/main.js:1
    updateSession moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/main.js:1
    subscription moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/main.js:1
    IA moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/main.js:1
    t moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    IA moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/main.js:1
    subscription moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/main.js:1
    y moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/vendor.js:2
    f moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/vendor.js:2
    _next moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/vendor.js:2
    next moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/vendor.js:2
    n moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/vendor.js:2
    _next moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/vendor.js:2
    next moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/vendor.js:2
    next moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/vendor.js:2
    n moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/vendor.js:2
    next moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/vendor.js:2
    next moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/vendor.js:2
    pushAccounts moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/main.js:1
    invoke moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    onInvoke Angular
    invoke moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    run moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    C moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    invokeTask moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    onInvokeTask Angular
    invokeTask moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    runTask moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
    b moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/polyfills.js:2
[vendor-angular.js:2:318527](moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/popup/vendor-angular.js)
Uncaught (in promise) TypeError: can't access dead object
    l moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/background.js:1
    o moz-extension://8e2124c7-5f5a-44d5-8b11-71234cc7acde/background.js:1
```

**Comment by iCodeHead** [comment]
Bitwarden Passwordless Firefox extension stops working after log in with device
Over 21 days ago I reported to Bitwarden support that the Bitwarden passwordless feature on Firefox extension stopped working after 'log in with device'. They responded that it is a known issue and are working on it. 
Has anyone else found a work around?

Test Conditions

    Windows 10 Version 10.0.19044 Build 19044 (01/2023) (Also tested on Win 8.1).

    Firefox version 110.0.1 & 111.0.0 & 111.0.1 (64-bit).

    Bitwarden 2023.2.1 Last updated March 7, 2023.

I installed the Bitwarden desktop app and logged in. In the Settings, turn on the 'Approve login requests' option in the Security section (off by default).

Steps To Reproduce

    Start with a clean Firefox profile using default settings.

    Install Bitwarden extension

    Login to the Bitwarden Firefox extension using 'Login with device' (not master password).

    Enter 2FA code if needed (I did not disable 2FA for testing).

    Close the extension.

    Re-open the extension again.

Result

    Bitwarden extension dropdown displays a blank screen and does not function but will still auto-fill on some sites.

Note: Disabling and re-enabling does not fix this issue for me.

This was also reported this on the www.reddit.com/r/Bitwarden site.
https://www.reddit.com/r/Bitwarden/comments/121y0q5/bitwarden_passwordless_firefox_extension_stops/

**Comment by MGibson1** [comment]
We've finally been able to reproduce this internally, and can start understanding the root cause. A big thank you to everyone who has posted reports, reproduction steps, and error logs. It's a huge help.

Early indications (thank you @DeveloperRob!) look like it's due to some Observables leaking into our background page from the popup view, but now that we can reproduce we can narrow down exactly which component is causing the problem.

**Comment by thechinchinsong** [comment]
Is there an update on this issue so far? I still have this issue on:

Windows 11 Home build: 10.0.2261 
FireFox version: 115.0.2 (64-bit)
Bitwarden extension version: 2023.5.1, Lasted Updated June 30, 2023

The same test conditions as @iCodeHead used above.

**Comment by DawsonFranz** [comment]
I can confirm I'm also having this issue as of August on the following platform:

Windows 11 Home build: 22621.1702
FireFox version: 116.0.2 (64-bit)
Bitwarden extension version: 2023.7.1

**Comment by dani-garcia** [comment]
Hey everyone, I've been going through some old issues to see which ones are still active and which might have already been fixed. 

I've been able to reproduce this issue with an older `2023.5.1` extension on Firefox, but I can't reproduce it anymore when using the latest published extension (`2024.6.2`). A lot of the internals of the extension have been recently reworked on the push to get the extension compatible with manifest V3, and we think those changes have indirectly fixed this issue as well.

I'd appreciate some confirmation if the issue is also fixed for you, or if it's still happening. Thanks!

**Comment by nighlabs** [comment]
@dani-garcia - I just tried device signin with Firefox Ext 2024.6.2 / Firefox 127.0 and had no issues.  🎉

In previous versions (apologies don't remember the versions) I had issues.

**Comment by Galactic-Hero** [comment]
Works for me now with version 2024.6.2 and current Firefox, Windows via iPhone Xr with current app -- Bill Fickas

    On Thursday, June 20, 2024 at 09:48:00 AM MDT, Daniel García ***@***.***> wrote:  
 
 


Hey everyone, I've been going through some old issues to see which ones are still active and which might have already been fixed.

I've been able to reproduce this issue with an older 2023.5.1 extension on Firefox, but I can't reproduce it anymore when using the latest published extension (2024.6.2). A lot of the internals of the extension have been recently reworked on the push to get the extension compatible with manifest V3, and we think those changes have indirectly fixed this issue as well.

I'd appreciate some confirmation if the issue is also fixed for you, or if it's still happening. Thanks!
